### PR TITLE
Check if the section sizes are valid ##dex

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

I'm trying to re-enable the fuzzed tests on CI (at least on master, probably not on every PR to not make tests too slow). While doing so I found there are issues with this clusterfuzz bin. @trufae what do you think? I'm no expert at all about dex, so if you think there's a better solution to fix this feel free to add commits on my branch or do a new PR or suggest how i should fix it.

On master:
```
[0x00000003]> iS
[Sections]

nth paddr                     size vaddr                    vsize perm name
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
0   0x00000000                0x70 0x00000000                0x70 -r-- header
1   0x00000070  0xffffffffffffff93 0x00000070  0xffffffffffffff93 -r-- constpool
2   0x00000003          0xe9caa270 0x00000003          0xe9caa270 -r-x code
3   0x000022e7               0xbee 0x000022e7               0xbee -r-- data
4   0x00000000              0x2ed5 0x00000000              0x2ed5 -r-- file
[0x00000003]> o
 3 * r-x 0x00002ed5 test/bins/fuzzed/clusterfuzz-testcase-minimized-ia_fuzz-5643065568526336.dms
[0x00000003]> om
 2 fd: 3 +0x00000003 0x00000003 - 0xe9caa272 r-x fmap.code
 1 fd: 3 +0x00000000 0x00000000 - 0x00002ed4 r-- fmap.file
```
As you can see the sections sizes are wrong and since those are used to map files, then we end up with a huge amount of data mapped, which makes analysis take forever.

**Test plan**

`radare2 -A test/bins/fuzzed/clusterfuzz-testcase-minimized-ia_fuzz-5643065568526336.dms `

### CODE REVIEW

File: libr/bin/p/bin_dex.c [hunk 1]
```diff
@@ -1926,34 +1926,36 @@ static RList *sections(RBinFile *bf) {
...
 			eprintf ("Warning: Invalid code size\n");
 			ptr->size = ptr->vaddr; // fix size
 		}
-		ptr->vsize = ptr->size;
-		// Commenting this line speedups loading from 4.5s to 3s
-		ptr->format = r_str_newf ("Cd %d[%d]", 4, ptr->vsize / 4);
-		ptr->perm = R_PERM_R;
-		ptr->add = false;
-		r_list_append (ret, ptr);
+		if (ptr->size <= bin->size) {
```
Comment (trufae) [inline libr/bin/p/bin_dex.c:1936]: ptr->size + ptr->paddr <= bin->size

File: libr/bin/p/bin_dex.c [hunk 2]
```diff
@@ -1926,34 +1926,36 @@ static RList *sections(RBinFile *bf) {
...
 	}
 	if ((ptr = R_NEW0 (RBinSection))) {
-		ptr->name = strdup ("code");
 		ptr->vaddr = ptr->paddr = bin->code_from; //ptr->vaddr = fsym;
 		ptr->size = bin->code_to - ptr->paddr;
-		ptr->vsize = ptr->size;
-		ptr->perm = R_PERM_RX;
-		ptr->add = false;
-		r_list_append (ret, ptr);
+		if (ptr->size <= bin->size) {
```
Comment (trufae) [inline libr/bin/p/bin_dex.c:1949]: same here

File: libr/bin/p/bin_dex.c [hunk 3]
```diff
@@ -1963,9 +1965,12 @@ static RList *sections(RBinFile *bf) {
 			// hacky workaround
 			//ptr->size = ptr->vsize = 1024;
 		}
-		ptr->perm = R_PERM_R; //|2;
-		ptr->add = false;
-		r_list_append (ret, ptr);
+		if (ptr->size <= bin->size) {
+			ptr->name = strdup ("data");
+			ptr->perm = R_PERM_R; //|2;
```
Comment (trufae) [inline libr/bin/p/bin_dex.c:1970]: remove comment

**Comment by trufae** [comment]
Screenshot 2020-06-20 at 12 24 34

**Comment by ret2libc** [comment]
@trufae I'm probably going to close this as I don't have the knowledge to work on this. I'm not familiar with the file format and I don't know how sections sizes are computed... Feel free to have a look if you want.

**Comment by trufae** [comment]
Can you open an issue and assign it to me? thanks!

**Pull request closed without merging**